provisioning/service/src/Authentication/ServiceConnectionString.cs

```diff
@@ -1,156 +1,235 @@
 {
     /// <summary>
-    /// This object handles the connection string for the Azure IoT Services.
+    /// The Service Connection String Parser class
     /// </summary>
-    /// <remarks>
-    /// The connection string contains a set of information that uniquely identify an IoT Service.
-    /// 
-    /// A valid connection string shall be in one of the following formats:
-    /// <code>
-    /// HostName=[ServiceName];SharedAccessKeyName=[keyName];SharedAccessKey=[Key]
-    /// HostName=[ServiceName];SharedAccessKeyName=[keyName];SharedAccessSignature=[Signature]
-    /// </code>
-    /// 
-    /// This object parse and store the connection string. It is responsible to provide the authorization token too. 
-    /// </remarks>
-    public sealed class ServiceConnectionString
+    public class ServiceConnectionString
     {
-        private static readonly TimeSpan DefaultTokenTimeToLive = TimeSpan.FromMinutes(5);
+        private const char ValuePairDelimiter = ';';
+        private const char ValuePairSeparator = '=';
+        private const string HostNameSeparator = ".";
+
+        private const string HostNamePropertyName = nameof(HostName);
+        private const string SharedAccessKeyNamePropertyName = nameof(SharedAccessKeyName);
+        private const string SharedAccessKeyPropertyName = nameof(SharedAccessKey);
+        private const string SharedAccessSignaturePropertyName = nameof(SharedAccessSignature);
+
+        private static readonly TimeSpan regexTimeoutMilliseconds = TimeSpan.FromMilliseconds(500);
+        private static readonly Regex HostNameRegex = new Regex(@"[a-zA-Z0-9_\-\.]+$", RegexOptions.Compiled | RegexOptions.IgnoreCase, regexTimeoutMilliseconds);
+        private static readonly Regex SharedAccessKeyNameRegex = new Regex(@"^[a-zA-Z0-9_\-@\.]+$", RegexOptions.Compiled | RegexOptions.IgnoreCase, regexTimeoutMilliseconds);
+        private static readonly Regex SharedAccessKeyRegex = new Regex(@"^.+$", RegexOptions.Compiled | RegexOptions.IgnoreCase, regexTimeoutMilliseconds);
+        private static readonly Regex SharedAccessSignatureRegex = new Regex(@"^.+$", RegexOptions.Compiled | RegexOptions.IgnoreCase, regexTimeoutMilliseconds);
+
+        private string _hostName;
+        private string _serviceName;
+
+        private ServiceConnectionString()
+        {
+        }
 
         /// <summary>
-        /// CONSTRUCOR
+        /// Factory for new Connection String object.
         /// </summary>
-        /// <param name="parser">the <see cref="ServiceConnectionStringParser"/> with the connection string content.</param>
-        /// <exception cref="ArgumentNullException">if the provided parser is null.</exception>
-        public ServiceConnectionString(ServiceConnectionStringParser parser)
+        /// <remarks>
+        /// The connection string contains a set of information that uniquely identify an IoT Service.
+        /// 
+        /// A valid connection string shall be in the following format:
```

Comment:
Please remove:
1. Avoid the "shall" wording (not really commonly used and shouldn't be part of our public documentation). Regardless, the formatting may change and our implementation should be resilient to addition so "shall" is too strong either way.
2. We should not specify the connection string. Instead, either point to the online spec if it exists or don't add it at all. The only place where one would get a connection string is from the Azure Portal. Devices should actually avoid them since it takes considerably more memory to parse (because of the RegEx situation above).